### DISCUSSION TITLE
How to edit authentication provider ID?

### DISCUSSION THREAD

**Discussion by pcolmer** [requarks/wiki#5404] [Help / Questions] [answered]
We're currently using Auth0 as the authentication system for our wiki.js system.

Within Auth0, we're adding an Enterprise connection to allow users to authenticate with their own IDP rather than username/password accounts in Auth0. That then generates a new provider ID. However, the user cannot access wiki.js because it gives this error:

```
insert into "users" ("createdAt", "email", "isActive", "isSystem", "isVerified", "localeCode", "name", "pictureUrl", "providerId", "providerKey", "tfaIsActive", "updatedAt") values ($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12) returning "id" - duplicate key value violates unique constraint "users_providerkey_email_unique"
```

So I think that we need to replace the old Auth0 provider ID with their new Auth0 provider ID so that they are then able to login. However, I cannot find a way to do this within the UI.

Can anyone provide any information as to where I might be able to find the information in the underlying database to hack it that way?

Thanks.

**Comment by NGPixel** [comment, marked as answer]
This cannot be done via the UI. You would need to modify the `users` DB table directly.

**Reply by pcolmer** [reply]
Thanks. I wasn't sure how hard it would be but the database schema is clear. Changing the data did not require a restart or downtime.
